### ISSUE TITLE
Confused with the softmax_bias

### ISSUE THREAD

**Issue by zen-star** [hustvl/SparseInst#57]
In sparseinst/decoder.py
``` python
iam_prob = F.softmax(iam.view(B, N, -1) + self.softmax_bias, dim=-1)
```

You add a learnable softmax_bias to iam, but it is identical to all elements as it is defined like ```self.softmax_bias = nn.Parameter(torch.ones([1, ]))```

softmax(x)_i = e_i / sum_j( exp(x_j) )
and softmax(x + c)_i = exp(x_i + c) / sum_j(exp(x_j + c)) = [ exp(c) * exp(x_i) ]  / [ exp(c) * sum_j(exp(x_j)) ] = e_i / sum_j( exp(x_j) )
that is softmax(x + c)_i = softmax(x + c)_i, which means the softmax_bias is useless.

I wonder if anything is misunderstood.

**Comment by wondervictor** [comment]
Hi @zen-star, you're right! Softmax with this bias is equivalent to the one without bias. The intuition of adding this bias is to adaptively and explicitly lower the output values of the instance activation maps and smooth the activations after softmax. Directly using softmax (e^x) will enable some pixels to dominate the activations and lose some contextual pixels. There are some other methods for smoothing the activations, e.g., softmax with temperature. For simplicity, we just add a learnable bias and it works. The final learned bias is negative and nearly -2.0, which is sound.

**Comment by zen-star** [comment]
Thanks.
However, the bias doesn't work whatever the learned value is. It is different from the softmax with temperature, which is the divisor. I suggest evaluating the pretrained model with any other bias value (e.g. 3.0) to see if the result is the same.
Look forward to your favourable reply.

**Comment by wondervictor** [comment]
Hi @zen-star, it might be a confusing setting. This bias works as an extra bias for the instance activation maps. Exactly, adding the bias will not affect the output of the softmax. Thank you for reminding me and I'll further check the details.

**Comment by zen-star** [comment]
Got it.
Thanks again for your elegant work^^
